Sources/ThresholdKey/Modules/TssSecurityQuestionModule.swift

```diff
@@ -0,0 +1,242 @@
+    ///   - newAnswer: The new answer for the security question.
+    ///   - answer: current answer
+    ///   - tag: tss tag
+    ///
+    /// - Returns: ``
+    ///
+    /// - Throws: `RuntimeError`, indicates invalid parameters was used or invalid threshold key or fail to change security question
+    public static func change_security_question( threshold : ThresholdKey, newQuestion: String, newAnswer: String, answer: String, tag: String) async throws {
+        let domainKey = TssSecurityQuestion + ":" + tag
+        
+        var isSet = false
+        do {
+            let question = try TssSecurityQuestionModule.get_question(threshold: threshold, tag: tag)
+            if question.count > 0 {
+                isSet = true
+            }
+        } catch {}
+        
+        if !isSet {throw "Security Question is not set"}
+        
+        // hash answer and new answer
+        guard let hash = answer.data(using: .utf8)?.sha3(.keccak256) else {
+            throw "Invalid answer format"
+        }
+
+        guard let newHash = newAnswer.data(using: .utf8)?.sha3(.keccak256) else {
+            throw "Invalid answer format"
+        }
+        
+        // get and decrypt factorkey and encrypt with new hash
+        guard let storeData = try threshold.get_general_store_domain(key: domainKey).data(using: .utf8) else {
+            throw "Invalid format"
+        }
+        var store = try JSONDecoder().decode(TssSecurityQuestionData.self, from: storeData)
+        
+        let associatedFactor = try decrypt(key: hash.toHexString(), msg: store.associatedFactor.toString() )
+        
+        let newHashKey = PrivateKey(hex: newHash.toHexString())
+        let newHashPub = try newHashKey.toPublic();
+        guard let encryptedData = try encrypt(key: newHashPub, msg: associatedFactor).data(using: .utf8) else {
+            throw "encryption error"
+        }
+        let encryptedMsg = try JSONDecoder().decode(EncryptedMessage.self, from: encryptedData)
+        
+        store.question = newQuestion
+        store.associatedFactor = encryptedMsg
+        
+        // set updated data to domain store
+        let jsonData = try JSONEncoder().encode(store)
+        
+        guard let jsonStr = String(data: jsonData, encoding: .utf8) else {
+            throw "Invalid security question data"
+        }
+        try threshold.set_general_store_domain(key: domainKey, data: jsonStr )
+        
+        try await threshold.sync_metadata();
+
+    }
+    
+    /// delete security question
+    /// - Parameters:
+    ///   - threshold_key: The threshold key to act on.
+    ///   - tag: tss tag
+    ///
+    /// - Returns: `String` public key of the factor
+    ///
+    /// - Throws: `RuntimeError`, indicates invalid parameters was used or invalid threshold key or fail to delete security question
+    public static func delete_security_question( threshold : ThresholdKey, tag: String) async throws -> String  {
```

Comment:
we should check if thesholdKey is initialized and has metadata key before allowing further.